osu.Game.Tests/Visual/Gameplay/TestSceneUnstableRateCounter.cs

```diff
@@ -0,0 +1,190 @@
+using osuTK;
+
+namespace osu.Game.Tests.Visual.Gameplay
+{
+    public class TestSceneUnstableRateCounter : OsuTestScene
+    {
+        [Cached(typeof(ScoreProcessor))]
+        private TestScoreProcessor scoreProcessor = new TestScoreProcessor();
+
+        [Cached(typeof(GameplayState))]
+        private GameplayState gameplayState;
+
+        [Cached(typeof(DrawableRuleset))]
+        private TestDrawableRuleset drawableRuleset = new TestDrawableRuleset();
+
+        private double prev;
+
+        public TestSceneUnstableRateCounter()
+        {
+            Score score = new Score
+            {
+                ScoreInfo = new ScoreInfo(),
+            };
+            gameplayState = new GameplayState(null, null, null, score);
+            scoreProcessor.NewJudgement += result => scoreProcessor.PopulateScore(score.ScoreInfo);
+        }
+
+        [SetUpSteps]
+        public void SetUp()
+        {
+            AddStep("Reset Score Processor", () => scoreProcessor.Reset());
+        }
+
+        [Test]
+        public void TestBasic()
+        {
+            AddStep("Create Display", () => recreateDisplay(new OsuHitWindows(), 5));
+
+            AddRepeatStep("Set UR to 250ms", () => setUR(25), 20);
+
+            AddStep("Reset UR", () =>
+            {
+                scoreProcessor.Reset();
+                recreateDisplay(new OsuHitWindows(), 5);
+            });
+
+            AddRepeatStep("Set UR to 100ms", () => setUR(10), 20);
+
+            AddStep("Reset UR", () =>
+            {
+                scoreProcessor.Reset();
+                recreateDisplay(new OsuHitWindows(), 5);
+            });
+
+            AddRepeatStep("Set UR to 0 (+50ms)", () => newJudgement(50), 20);
+
+            AddStep("Reset UR", () =>
+            {
+                scoreProcessor.Reset();
+                recreateDisplay(new OsuHitWindows(), 5);
+            });
+
+            AddRepeatStep("Set UR to 0 (-50ms)", () => newJudgement(-50), 20);
+
+            AddRepeatStep("New random judgement", () => newJudgement(), 40);
+        }
+        private void recreateDisplay(HitWindows hitWindows, float overallDifficulty)
+        {
+            hitWindows?.SetDifficulty(overallDifficulty);
+
+            drawableRuleset.HitWindows = hitWindows;
+
+            Clear();
+
+            Add(new UnstableRateCounter
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                Scale = new Vector2(5),
+            });
+
+            Add(new BarHitErrorMeter
+            {
+                Anchor = Anchor.BottomCentre,
+                Origin = Anchor.CentreLeft,
+                Rotation = 270,
+            });
+
+            Add(new ColourHitErrorMeter
+            {
+                Anchor = Anchor.BottomCentre,
+                Origin = Anchor.CentreLeft,
+                Rotation = 270,
+                Margin = new MarginPadding { Left = 50 }
+            });
```

Comment:
I originally left them there so that it would be visible at which offset the notes were landing, but I have now gone ahead and removed it as you suggested. This also removes the TestDrawableRuleset class that you mentioned below.